lib/OpusVL/FB11X/SysParams/Controller/SysParams.pm

```diff
@@ -21,160 +21,66 @@ __PACKAGE__->config
 {
-    my ($self, $c, $name, $textarea) = @_;
-    $self->add_final_crumb($c, 'Edit');
-	my $param = $c->model ('SysParams::SysInfo')->find({name => $name});
-    $c->detach('/not_found') unless $param;
-
-    my $value = $param->value;
-    my $label = $param->label // $param->name;
-	my $return_url = $c->stash->{urls}{sys_info_list}->();
-
-	if ($c->req->param ('cancelbutton')) {
-		$c->flash->{status_msg} = 'System parameter not changed';
-		$c->res->redirect ($return_url);
-		$c->detach;
-	}
-
-    my $form = $self->param_edit_form($param);
-    $c->stash->{form} = $form;
-    $c->stash->{param} = $param;
-
-	if ($form->process( params => $c->req->params )) {
-        $c->flash->{status_msg} = 'System Parameter Successfully Altered';
-		$c->res->redirect($return_url);
-		$c->detach;
-	}
-}
+    my ($self, $c, $name) = @_;
+    my $manager = OpusVL::FB11::Hive
+        ->service('sysparams::management')
+        ->for_all_components;
 
-sub del_param
-	: Path('del')
-	: Args(1)
-    : FB11Feature('System Parameters')
-{
-    my ($self, $c, $param) = @_;
-	my $p = $c->model ('SysParams::SysInfo')->find({name => $param});
-    $c->detach('/not_found') unless $p;
-	my $value = $p->value;
-
-    my $param_name  = $p->label // $p->name;
-    $c->stash->{question} = "Are you sure you want to delete the parameter: ${param_name}";
-    $c->stash->{template} = 'fb11/admin/confirm.tt';
-    $c->stash->{form} = $self->form($c, '+OpusVL::FB11::Form::Confirm');
-    my $form = $c->stash->{form};
-    $form->process($c->req->params);
-	my $return_url = $c->stash->{urls}{sys_info_list}->();
-
-	if ($form->validated) {
-        if ($c->req->params->{submitok}) {
-		    $c->model('SysParams::SysInfo')->del ($param);
-		    $c->flash->{status_msg} = 'System Parameter Successfully Deleted';
-		    $c->res->redirect ($return_url);
-		    $c->detach;
-        }
-        else {
-            $c->flash->{status_msg} = 'System Parameter Not Deleted';
-            $c->res->redirect ($return_url);
-            $c->detach;
-        }
-	}
-}
+    my $meta = $manager->metadata_for($name) // $c->detach('/not_found');
+    my $value = $manager->value_of($name);
 
-sub new_param
-	: Path('new')
-	: Args(0)
-    : FB11Feature('System Parameters')
-{
-	my $self  = shift;
-	my $c     = shift;
+    my $label = $meta->{label};
 
-	my $return_url = $c->stash->{urls}{sys_info_list}->();
-    my $param = $c->model('SysParams::SysInfo')->new_result({});
+    if ($c->req->param ('cancelbutton')) {
+        $c->flash->{status_msg} = 'System parameter not changed';
+        $c->res->redirect($c->uri_for($self->action_for('list_params')));
+        $c->detach;
+    }
 
-    my $form = $self->param_edit_form($param);
+    my $form = $self->edit_form;
     $c->stash->{form} = $form;
-    $c->stash->{param} = $param;
 
-    my $ok = try {
-        $form->process( 
-            params => $c->req->params,
-            init_object => { data_type => 'text' }
-        )
+    $form->process(
```

Comment:
Does $form->process by any chance lean on it being DBIC?

Reply:
This part of the controller is now a 404 until I can get around to implementing it properly.